airflow/www/views.py

```diff
@@ -4375,6 +4377,18 @@ def process_form(self, form, is_created):
                     # value isn't an empty string.
                     if value != "":
                         extra[field_name] = value
+                    field = form._fields.get(key)
+                    # Check if the field is a BS3TextFieldWidget
+                    if isinstance(field.widget, BS3TextFieldWidget):
+                        value = form.data.get(key, "")
+                        if isinstance(value, str):
+                            value = value.strip()
+                            if value:
```

Comment:
Isn't this warning users even if the field data has not been deleted? I don't think we need to show a warning when a user modifies an existing value. If it's not possible to determine whether a field was previously populated, I think we should just leave do nothing.

Reply:
Yes, it does warn users even if the field data has not been deleted. I couldn't find how to process the form when the 'Save' button is clicked and I could only find how to process it after the button is clicked in the codebase. If that's the case, would it be better to go with the general warning and get rid of this?

Reply:
Based on our discussion, I've kept the general warning on the connection edit form.